### PR TITLE
Add more missing methods to `os._Environ`

### PR DESCRIPTION
**Pull request by AlexWaygood**
```python
>>> from os import environ
>>> _Environ = type(environ)
>>> _Environ.__or__
<function _Environ.__or__ at 0x0000029ABDE881F0>
>>> _Environ.__ror__
<function _Environ.__ror__ at 0x0000029ABDE88280>
>>> _Environ.__class_getitem__
<bound method GenericAlias of <class 'os._Environ'>>
>>> type(environ | environ)
<class 'dict'>
>>> from collections import OrderedDict
>>> type(environ | OrderedDict([('foo', 'bar')]))
<class 'dict'>
>>> _Environ[int, str]
os._Environ[int, str]
```

### CODE REVIEW

File: stdlib/os/__init__.pyi [hunk 1]
```diff
@@ -246,9 +248,13 @@ class _Environ(MutableMapping[AnyStr, AnyStr], Generic[AnyStr]):
     def __iter__(self) -> Iterator[AnyStr]: ...
     def __len__(self) -> int: ...
     if sys.version_info >= (3, 9):
+        def __class_getitem__(cls, __item: Any) -> GenericAlias: ...
+        def __or__(self, __value: Mapping[_T1, _T2]) -> dict[AnyStr | _T1, AnyStr | _T2]: ...
+        def __ror__(self, __value: Mapping[_T1, _T2]) -> dict[AnyStr | _T1, AnyStr | _T2]: ...
         # We use @overload instead of a Union for reasons similar to those given for
         # overloading MutableMapping.update in stdlib/typing.pyi
-        @overload
+        # type: ignore is needed due to incompatible __or__/__ior__ signatures
```
Comment (AlexWaygood) [inline stdlib/os/__init__.pyi:256]:
```suggestion
        # The type: ignore is needed due to incompatible __or__/__ior__ signatures
```

**Comment by AlexWaygood** [comment]
mypy doesn't like having incompatible `__or__` and `__ior__` methods. I've add `# type: ignore` comments because I think @posita's reasoning for keeping the `__ior__` signature the same as `dict.update` is solid, so I don't want to change the `__ior__` signature for the sake of satisfying mypy.

**Comment by posita** [comment]
Oh! Good catch. I _think_ I barely tested wether 'os.environ | {}' passed validation after #6921 (and then only on Python 3.9), _not_ what the resulting type was. Thanks for this!

File: stdlib/os/__init__.pyi [hunk 2]
```diff
@@ -246,9 +248,13 @@ class _Environ(MutableMapping[AnyStr, AnyStr], Generic[AnyStr]):
     def __iter__(self) -> Iterator[AnyStr]: ...
     def __len__(self) -> int: ...
     if sys.version_info >= (3, 9):
+        def __class_getitem__(cls, __item: Any) -> GenericAlias: ...
```
Comment (Akuli) [inline stdlib/os/__init__.pyi:251]: Why is this here? I don't see it in the implementation.

Comment (AlexWaygood) [inline stdlib/os/__init__.pyi:251]:
```suggestion
```

Comment (AlexWaygood) [reply stdlib/os/__init__.pyi]: You're right, it's inherited from `collections.abc.MutableMapping` -- I forgot that classmethods became bound to the class even if you accessed them from the class, so did `type(environ).__class_getitem__ is MutableMapping.__class_getitem__` as a test, which evaluated to `False`... etc. etc...

Comment (AlexWaygood) [reply stdlib/os/__init__.pyi]: Although we don't actually define it on `MutableMapping` in typeshed due to the fact that we pretend it's the same class as `typing.MutableMapping`... But that's a subject for another issue (and probably not really worth fussing about, anyway).
